### DISCUSSION TITLE
How can I find out which diode or switch has bad connection when it's not working under the test firmware

### DISCUSSION THREAD

**Discussion by FebriCN** [Yowkees/keyball#657] [Q&A]
When I use the test firmware, I've found some random keys not working (not print letters or keyboard's OLED screen not showing the key name).

I'm very very new to soldering things or KiCAD, how can I find out the actual problem?

Based on my very few knowledge, since it's random keys not responding, I'm guessing the problem is **on switches** _not diodes_, am I right?

**Comment by Yowkees** [comment]
Hi.

Usually PCB sockets or diodes have soldering problem in this case.

If you can see KiCAD, just tracking which diodes is connected to problem switches.

Best,
